.github/workflows/ruby.yml

```diff
@@ -12,9 +12,18 @@ jobs:
     - name: Set up Ruby 2.7
       uses: actions/setup-ruby@v1
       with:
-        ruby-version: 2.7.1
-    - name: RuboCop
+        ruby-version: 2.7
+    - name: Cache gems
+      uses: actions/cache@v1
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-rubocop-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-rubocop-
+    - name: Install gems
       run: |
-        gem install bundler
+        bundle config path vendor/bundle
+        bundle config set without 'pages'
```

Comment:
Why is this config necessary?

Reply:
To get the major speed gain.

From [the commit](https://github.com/rails/rails/commit/100f7a7f643ae6766f406ccf1e049050199caddc) mentioned in the description:  
>Unfortunately it's not possible to tell Bundler to only install gems
from a single group, so we have to tell it not to install every other
group instead.

Any proposal to improve this?

Reply:
I understand we don't need to install gems under the `pages` group, but what about specifying the `--without=pages` flag to `bundle install` instead for simplicity?

Reply:
I see no difference in this case. OK, changed.  

From https://github.com/Shopify/ruby-style-guide/pull/159/checks?check_run_id=733139479:  
```
Run bundle install --jobs 4 --retry 3 --without pages --path vendor/bundle
[DEPRECATED] The `--path` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set path 'vendor/bundle'`, and stop using this flag
[DEPRECATED] The `--without` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set without 'pages'`, and stop using this flag
```